50k GitHub stars blog (#10)
localstack/localstack-blog.github.io: content/2023-11-22-celebrating-50k-github-stars-localstack/index.md

LocalStack, originally a side project, first garnered public attention with a post on [Hacker News](https://news.ycombinator.com/item?id=13966088) (also archived on the [Internet Archive](https://web.archive.org/web/20231122065745/https://news.ycombinator.com/item?id=13966088)). This post introduced the concept of a fully local mocking/testing framework for cloud applications, drawing the interest of the broader tech community, including notable AWS figures. A key moment in LocalStack's journey was when Jeff Barr, Chief evangelist at AWS, [shared a tweet about LocalStack](https://twitter.com/jeffbarr/status/846382903210663936) (also archived on the [Internet Archive](https://chat.openai.com/c/c591ffcf-bd80-4de2-a6b1-ac5cf13d2903)), leading to an overnight surge in GitHub stars, with thousands joining our community.

Here is a graphic that captures the watershed moment, and the subsequent growth of LocalStack as a project:

<graphic>

## Engaging with the community

> The community's response was overwhelmingly positive — developers began actively [contributing to LocalStack](https://github.com/localstack/localstack/graphs/contributors), [creating issues](https://github.com/localstack/localstack/issues), [providing suggestions](https://discuss.localstack.cloud/), and [submitting pull requests](https://github.com/localstack/localstack/pulls). This engagement wasn't just from individual developers — entire teams from various companies, specifically those focused on developer experience, started using LocalStack to create efficient development environments. 

Reviewer: I kind of like them tbh, what would you use instead @MarcelStranak - commas, or colons?

> The community's response was overwhelmingly positive — developers began actively [contributing to LocalStack](https://github.com/localstack/localstack/graphs/contributors), [creating issues](https://github.com/localstack/localstack/issues), [providing suggestions](https://discuss.localstack.cloud/), and [submitting pull requests](https://github.com/localstack/localstack/pulls). This engagement wasn't just from individual developers — entire teams from various companies, specifically those focused on developer experience, started using LocalStack to create efficient development environments. 

Reviewer 2: Colon looks better in this case, since we are listing the items. Additionally, we could cut it into two sentences and use a different tone.
Anyhow, I am here only for the proofreading, I leave the writing to Harsh. ;)

> Going next, we will continue to focus on strengthening the LocalStack ecosystem and integrations with frameworks such as *[Testcontainers](https://testcontainers.com/), [CloudFormation](https://docs.localstack.cloud/user-guide/aws/cloudformation/), [CDK](https://docs.localstack.cloud/user-guide/integrations/cdk-for-terraform/)* and more! With the new [Extension mechanism](https://docs.localstack.cloud/user-guide/extensions/), you can now extend and customize LocalStack to your needs. We have shipped official extensions for [Minilfare](https://miniflare.dev/), [Mailhog](https://github.com/mailhog/MailHog), [Stripe](https://github.com/adrienverge/localstripe), [httpbin](https://httpbin.org/), and more to provide you with a smooth local development experience.

Reviewer: ```suggestion
Going next, we will continue to focus on strengthening the LocalStack ecosystem and integrations with frameworks such as *[Testcontainers](https://testcontainers.com/), [CloudFormation](https://docs.localstack.cloud/user-guide/aws/cloudformation/), [CDK](https://docs.localstack.cloud/user-guide/integrations/cdk-for-terraform/)* and more! With the new [Extension mechanism](https://docs.localstack.cloud/user-guide/extensions/), you can now extend and customize LocalStack to your needs. We have shipped official extensions for [Miniflare](https://miniflare.dev/), [Mailhog](https://github.com/mailhog/MailHog), [Stripe](https://github.com/adrienverge/localstripe), [httpbin](https://httpbin.org/), and more to provide you with a smooth local development experience.
```

The best is yet to come! We are primarily working on making our users happy with new features and improving LocalStack as much as we can do. With the [3.0 release](https://blog.localstack.cloud/2023-11-16-announcing-localstack-30-general-availability/), LocalStack has evolved from being an open-source project to an enterprise development framework! We have also received a lot of feedback from our individual users, and we've heard you loud and clear — We're super excited to bring our **Hobby Plan** for individuals *who are not using LocalStack commercially*!

We want to **lower the barrier of entry** for everybody and make it **easier to get started**. With the Hobby Plan, we want to enable more users to use the LocalStack emulator with all its otherwise **paid features**. LocalStack's Hobby Plan allows you to get started with:

- [Advanced AWS services](https://docs.localstack.cloud/user-guide/aws/feature-coverage/)
- [CI Starter Tier](https://docs.localstack.cloud/user-guide/ci/ci-keys/)
- [IAM Policy Enforcement](https://docs.localstack.cloud/user-guide/security-testing/iam-enforcement/)
- [State Persistence](https://docs.localstack.cloud/references/persistence-mechanism/)

> LocalStack is deeply rooted in open source and by making LocalStack available for free for non-commercial projects we hope to **further support and give back to the open source community**. It is the easiest way to get into cloud development and to learn not only the specific AWS services but also general cloud concepts that are also applicable to all the other cloud platforms out there. Using the Hobby Plan does not require a credit card, no need to set up an AWS account and there is also no cost risk (or using up the **AWS Free Tier**).

Reviewer 2: ```suggestion
LocalStack is deeply rooted in open-source, and by making LocalStack available for free for non-commercial projects, we hope to **further support and give back to the open-source community**. It is the easiest way to get into cloud development and to learn not only the specific AWS services but also general cloud concepts that are also applicable to all the other cloud platforms out there. Using the Hobby Plan does not require a credit card. There is no need to set up an AWS account, and there is also no cost risk (or use up the **AWS Free Tier**).
```

*Get started with LocalStack Hobby Plan today! [Sign up here](https://app.localstack.cloud/pricing/).*

## Conclusion

> To put it in a nutshell — We are incredibly thankful to our community for all the support and feedback we have received over the years! Your contributions have made LocalStack, a battle-tested cloud emulation platform, and helped us evolved from a project that leveraged existing tools to a more integrated platform, capable of emulating complex interactions between various AWS services.

Reviewer 2: ```suggestion
To put it in a nutshell, we are incredibly thankful to our community for all the support and feedback we have received over the years! Your contributions have made LocalStack a battle-tested cloud emulation platform. They helped us evolve from a project that leveraged existing tools to a more integrated platform that is capable of emulating complex interactions between various AWS services.
```

> 

Reviewer: Would maybe add a short paragraph somewhere in here that mentions something along the lines of:

> Our longer-term goal is not only to provide the best platform and tooling for local cloud development, but to shape the mindset of the industry as a whole and spearhead a movement that puts developer experience (DevX) at the forefront of all cloud-based and managed service offerings. With the groundbreaking and pioneering work that the LocalStack team and our community are conducting on a daily basis, we have demonstrated that it is indeed possible to fundamentally re-think the way how cloud applications are developed. Some cloud providers like Cloudflare have already embarked on a [local-first API development](https://developers.cloudflare.com/workers/observability/local-development-and-testing/) journey, and we anticipate that others will follow the lead. LocalStack will be in a prime position to offer a unified developer platform that facilitates local multi-cloud development across various providers and managed services - metaphorically becoming the central "app store" for local cloud developer tooling.

Unsure about the last sentence, could probably also be removed/revised, or we find a better metaphor..

> Our mission remains the same — To give developers back control over their environments and free them from wasting time with inefficient dev&test loops in the cloud, so they can instead focus on developing great products! We are excited about making LocalStack more configurable and adaptable to specific user needs, and we are looking forward to your continued support and feedback!

Reviewer 2: ```suggestion
Our mission remains the same: Give developers back control over their environments and stop them from wasting time on inefficient development and testing loops in the cloud, so they can focus on creating great products! We are excited about making LocalStack more configurable and adaptable to specific user needs, and we are looking forward to your continued support and feedback!
```

The best is yet to come! We are primarily working on making our users happy with new features and improving LocalStack as much as we can do. With the [3.0 release](https://blog.localstack.cloud/2023-11-16-announcing-localstack-30-general-availability/), LocalStack has evolved from being an open-source project to an enterprise development framework! We have also received a lot of feedback from our individual users, and we've heard you loud and clear — We're super excited to bring our **Hobby Plan** for individuals *who are not using LocalStack commercially*!

We want to **lower the barrier of entry** for everybody and make it **easier to get started**. With the Hobby Plan, we aim to enable more users to use the LocalStack emulator with all its otherwise **paid features**. LocalStack's Hobby Plan allows you to get started with:

- [Advanced AWS services](https://docs.localstack.cloud/user-guide/aws/feature-coverage/)
- [CI Starter Tier](https://docs.localstack.cloud/user-guide/ci/ci-keys/)
- [IAM Policy Enforcement](https://docs.localstack.cloud/user-guide/security-testing/iam-enforcement/)
- [State Persistence](https://docs.localstack.cloud/references/persistence-mechanism/)

> LocalStack is deeply rooted in open-source, and by making LocalStack available for free for non-commercial projects, we hope to **further support and give back to the open-source community**. It is the easiest way to get into cloud development and to learn not only the specific AWS services but also general cloud concepts that are also applicable to all the other cloud platforms out there. Using the Hobby Plan does not require a credit card. There is no need to set up an AWS account, and there is also no cost risk (or use up the **AWS Free Tier**).

Reviewer: ```suggestion
LocalStack is deeply rooted in open-source, and by making LocalStack available for free for non-commercial projects, we hope to **further support and give back to the open-source community**. It is the easiest way to get into cloud development and to learn not only the specific AWS services but also general cloud concepts that are also applicable to all the other cloud platforms out there. Using the Hobby Plan does not require a credit card. There is no need to set up an AWS account, and there is also no cost risk (as with using up the AWS Free Tier).
```

*Get started with LocalStack Hobby Plan today! [Sign up here](https://app.localstack.cloud/pricing/).*

## Conclusion

> To put it in a nutshell, we are incredibly thankful to our community for all the support and feedback we have received over the years! Your contributions have made LocalStack a battle-tested cloud emulation platform. They helped us evolve from a project that leveraged existing tools to a more integrated platform that is capable of emulating complex interactions between various AWS services.

Reviewer: ```suggestion
To put it in a nutshell, we are incredibly thankful to our community for all the support and feedback we have received over the years! Your contributions have made LocalStack a battle-tested cloud emulation and development platform. They helped us evolve from a project that leveraged existing tools to a full-fledged platform that is capable of providing an integrated local development environment for building AWS cloud and serverless applications.
```

Our mission remains the same: Give developers back control over their environments and free them from wasting time on inefficient development & testing loops in the cloud, so they can focus on creating great products! We are excited about making LocalStack more configurable and adaptable to specific user needs, and we are looking forward to your continued support and feedback!

> *Onto hitting the 100,000 GitHub Stars milestone next!* 🚀

Reviewer: ```suggestion
*_Ad astra_ - onto hitting the 100,000 GitHub Stars milestone next!* 🚀
```

> Our mission remains the same: Give developers back control over their environments and free them from wasting time on inefficient development & testing loops in the cloud, so they can focus on creating great products! We are excited about making LocalStack more configurable and adaptable to specific user needs, and we are looking forward to your continued support and feedback!

Reviewer: ```suggestion
Our mission remains the same: Give developers back control over their environments and free them from wasting time on inefficient development & testing loops in the cloud, so they can focus on creating great products. We are excited about making LocalStack more configurable and adaptable to specific user needs, and we are looking forward to your continued support and feedback!
```
